### DISCUSSION TITLE
Import cables from CSV, what am I missing?

### DISCUSSION THREAD

**Discussion by lonelyadmin** [netbox-community/netbox#5791] [Q&A] [answered]
I don't understand what the import tool expects to have, am I missing something simple? I can't find much of anything referencing imports in the docs. I have thousands of "cables" to create that I have defined manually in a csv ready to go. Just connecting rear ports of patch panels between buildings.

Where can I find what fields the import tool expects and how can I validate the data types that it will allow? 

Exporting cables to CSV gives me:

termination_a_type | termination_a_id | termination_b_type | termination_b_id | type | status | label | color | length | length_unit
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
dcim.interface | 56 | dcim.frontport | 7 | Singlemode Fiber | Connected | test_label | ffeb3b |   |  

Import from CSV shows:

side_a_device | side_a_type | side_a_name | side_b_device | side_b_type | side_b_name
-- | -- | -- | -- | -- | --

Using v2.10.4.

**Comment by candlerb** [comment, marked as answer]
I'm afraid that importing and exporting CSV are not compatible with each other - they use different columns and different values. See [this thread](https://groups.google.com/g/netbox-discuss/c/EAHh0_PRNno/m/pP-j0gM5AAAJ).

On Netbox v2.10 you can simply click on the question mark next to the column names in the import page, to get the available choices.  For patch panel rear ports this would be `dcim.rearport`.  The `side_a_device` is the *name* of the device, and the `side_a_name` is the *name* of the rearport (not the database ID).

You can import type, status, label etc as well if you want - just add the extra columns.  In the CSV import page, under Netbox 2.10, the question mark icon to the right of the field will give you the allowed values.

**Reply by lonelyadmin** [reply]
Thanks for that explanation, helped me see the light!

**Comment by TigerEnteNBG** [comment]
Normally netbox gives you an error whats wrong or missing.
Here's an expamle line I used some days ago:

side_a_device,side_a_type,side_a_name,side_b_device,side_b_type,side_b_name,type,label,color,length,length_unit
DeviceA,dcim.rearport,Rear-5,DeviceB,dcim.frontport,Front-5,cat7,Lable,bebebe,3,m

**Reply by lonelyadmin** [reply]
Thanks for the example, with these two replies I was able to get all the cables added.
